### ISSUE TITLE
blank screen with thunderbolt connected monitor

### ISSUE THREAD

**Issue by croots** [hyprwm/Hyprland#7680]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
<Paste the output of the command here>
```

</details>

### Description

External monitor connected via thunderbolt (from my laptop directly to the monitor) does not consistently display and spawn a workspace when it is hotplugged. Works when Hyprland is started with the monitor plugged in. Regression from 41 when this worked reliably and regression from 42 (non -git release) when it worked inconsistently.  Peripherals and charging correctly work.

Symptoms matched  #7321 but issue was not fixed when others were.

### How to reproduce

Observe display on boot, unplug Dell XPS 9310 from an HP monitor via thunderbolt, plug back in, observe no display.

### Crash reports, logs, images, videos

Logs forthcoming based on hyprwm/aquamarine#76

**Comment by croots** [comment]
[dmesg.log](https://github.com/user-attachments/files/16913006/dmesg.log)

**Comment by croots** [comment]
[hyprland.log](https://github.com/user-attachments/files/16913105/hyprland.log)

**Comment by romanstingler** [comment]
can you please add your hyprland.conf

**Comment by ikalco** [comment]
hmm.. does it work on sway?

btw this seems unrelated to AQ_NO_MODIFIERS since there is no `DDB` stuff in dmesg.log

**Comment by croots** [comment]
@romanstingler  Sorry for the delay, this setup is at my workplace.

[env.txt](https://github.com/user-attachments/files/16933163/env.txt)

Most of my dotfiles are from Illogical Impulse (though I've since moved on to an amalgamation of a few setups so there's some customization)

[hyprland.conf.txt](https://github.com/user-attachments/files/16933348/hyprland.conf.txt)
[env.conf.txt](https://github.com/user-attachments/files/16933275/env.conf.txt)
[execs.conf.txt](https://github.com/user-attachments/files/16933317/execs.conf.txt)
[general.conf.txt](https://github.com/user-attachments/files/16933237/general.conf.txt)
[rules.conf.txt](https://github.com/user-attachments/files/16933363/rules.conf.txt)
[colors.conf.txt](https://github.com/user-attachments/files/16933364/colors.conf.txt)
[keybinds.conf.txt](https://github.com/user-attachments/files/16933365/keybinds.conf.txt)

**Comment by croots** [comment]
Update - When I boot the machine up unplugged and then plug it in, the display works fine. It seems as though if the display has been seen already since first Hyprland launch it fails to get readded.

**Comment by ikalco** [comment]
> Update - When I boot the machine up unplugged and then plug it in, the display works fine. It seems as though if the display has been seen already since first Hyprland launch it fails to get readded.

uuuh... can you give hyprland.log where you boot, and replug so it works

**Comment by Estebiu** [comment]
Any updates? I'm still having this exact same problem.

**Comment by c-danil0o** [comment]
I am using workaround for this where I just restart the monitor and then image appears.

**Comment by christophe-vescovi** [comment]
I have the same issue. What is working for me is :
```
hyprctl dispatch dpms off DP-x
hyprctl dispatch dpms on DP-x
```
Or two dpms toggle...

**Comment by erahhal** [comment]
> I have the same issue. What is working for me is :
> ```
> hyprctl dispatch dpms off DP-x
> hyprctl dispatch dpms on DP-x
> ```
> Or two dpms toggle...

I ended up with the same hack myself.

**Comment by ikalco** [comment]
@christophe-vescovi @erahhal
can you guys give a full hyprlandd.log of you guys using the dpms trick
I think the problem is that the enable commit fails because a pageflip is happening or something

**Comment by erahhal** [comment]
I've attached the log.
[hyprland.log](https://github.com/user-attachments/files/17414946/hyprland.log)

**Comment by erahhal** [comment]
Not sure if related but on occasion the DPMS trick doesn't work, and when this happens `wdisplays` and `wlr-randr` show the same external display twice.

**Comment by croots** [comment]
> > Update - When I boot the machine up unplugged and then plug it in, the display works fine. It seems as though if the display has been seen already since first Hyprland launch it fails to get readded.
> 
> uuuh... can you give hyprland.log where you boot, and replug so it works

Hi! Sorry for the delay, I've been in super crunch mode since September. Here's a log file of me launching hyprland, Plugging in my monitor successfully, unplugging, plugging in my monitor unsuccessfully, unplugging, then closing hyprland.

[hyprland.log](https://github.com/user-attachments/files/17561180/hyprland.log)

**Comment by ikalco** [comment]
@croots can you try this [patch.txt](https://github.com/user-attachments/files/17580545/patch.txt) for hyprland

@erahhal your issue seems unrelated based on logs (its some weird output manager stuff like you said)
you should make an issue and try to find a minimum repo with a config as close to the default one as possible

**Comment by croots** [comment]
> @croots can you try this [patch.txt](https://github.com/user-attachments/files/17580545/patch.txt) for hyprland

This patch worked with 93b4478e70af6ffb08a4a66a6d0364c3296db296! Thanks for your help.

**Comment by sillen102** [comment]
I have this exact issue on 0.46.2

**Comment by ikalco** [comment]
> I have this exact issue on 0.46.2

update Hyprland and Aquamarine versions, many issues like this fixed there

**Comment by sillen102** [comment]
> > I have this exact issue on 0.46.2
> 
> update Hyprland and Aquamarine versions, many issues like this fixed there

I did now. Built from source, commit 8a6778f0a087cdfc4bc1d3751b0be2c2bf3322aa. It still doesn't work, but differently. Now the monitor is not blank but I can't get it to display in the correct resolution. I get the error that the monitor failed to set any requested modes and then it defaults to 1920x1080 (it's a 4K monitor).

Strangely enough it does work if I disable my internal monitor by running: `hyprctl --batch 'keyword monitor eDP-1,disable'`
then the 4K monitor gets it's proper resolution, but now my internal monitor is disabled and I cannot enable it again as it stays blank even after I unplug the cable. I tried setting up a script that I could trigger with a keybinding that runs: `hyprctl --batch 'keyword monitor eDP-1,3840x2160,auto,2'` but it doesn't work. I have to kill Hyprland and log in again to get the internal screen back.
